pkg/didmethod/peer/did.go

```diff
@@ -30,7 +32,59 @@ func newDid(doc *did.Doc) (string, error) {
 	if doc.PublicKey == nil || doc.Authentication == nil {
 		return "", errors.New("the genesis version must include public keys and authentication")
 	}
+	numBasis, err := hashString(doc)
+	if err != nil {
+		return "", err
+	}
+	messageIdentifier := []string{peerPrefix, numAlgo, encAlgo, "-", numBasis}
+	peerDID := strings.Join(messageIdentifier, "")
+
+	return peerDID, nil
+}
+
+func validate(doc *did.Doc) error {
+	did := doc.ID
+	hashedNumBasis, err := validateDid(did)
+	if err != nil {
+		return err
+	}
+	//removing did to calculate the hash of genesis doc
+	doc.ID = ""
+	numBasis, err := hashString(doc)
+	if err != nil {
+		return err
+	}
+
+	if !(numBasis == hashedNumBasis) {
+		return errors.New("hash of the doc doesnt match the computed hash")
+	}
+	return nil
+}
+
+func validateDid(did string) (string, error) {
+	//no whitespace
+	for _, v := range did {
+		// test each character to see if it is whitespace.
+		if unicode.IsSpace(v) {
+			if v == 32 {
```

Comment:
I think this `if` check is not required given the use of `unicode.IsSpace()`

Reply:
removed